scripts/Diffraction/isis_powder/routines/sample_details.py

```diff
@@ -146,6 +195,46 @@ def thickness(self):
         else:
             raise RuntimeError("Thickness is not applicable for the shape type \"{}\"".format(self._shape_type))
 
+    def generate_sample_geometry(self):
+        """
+        Generates the expected input for sample geometry using the SampleDetails class
+        :param self: Instance of SampleDetails containing details about sample geometry and material
+        :return: A map of the sample geometry
+        """
+        return self._shape.generate_sample_geometry()
+
+    def generate_sample_material(self):
+        """
+        Generates the expected input for sample material using the SampleDetails class.
+        See SetSampleMaterial for documentation on this dictionary
+        :param self: Instance of SampleDetails containing details about sample geometry and material
+        :return: A map of the sample material
+        """
+        material_json = {'ChemicalFormula': self.material_object.chemical_formula}
+        if self.material_object.number_density:
+            material_json["SampleNumberDensity"] = self.material_object.number_density
+        if self.material_object.absorption_cross_section:
+            material_json["AttenuationXSection"] = self.material_object.absorption_cross_section
+        if self.material_object.scattering_cross_section:
+            material_json["ScatteringXSection"] = self.material_object.scattering_cross_section
+        return material_json
+
+    def generate_container_geometry(self):
+        if self._container_shape:
+            return self._container_shape.generate_container_geometry()
+        else:
+            return None
+
+    def generate_container_material(self):
```

Comment:
You still need this if else statement around the rest of the lines in case no container is set:
```
if self.container_material_object:
...
else:
   return None
```
This prevents an error if you do a PP run without having called sample_details.set_container